google/resumable_media/_helpers.py

```diff
@@ -131,4 +183,19 @@ def http_request(transport, method, url, data=None, headers=None):
     Returns:
         object: The return value of ``transport.request()``.
     """
-    return transport.request(method, url, data=data, headers=headers)
+    response = transport.request(method, url, data=data, headers=headers)
+    if get_status_code(response) not in RETRYABLE:
+        return response
+
+    total_sleep = 0.0
+    num_retries = 0
+    while total_sleep < MAX_CUMULATIVE_RETRY:
```

Comment:
it seems like this retry logic is "unrelated" to the request object, so I'd prefer to pull it into its own thing, maybe something (conceptually) like:

```python
func = functools.partial(transport.request, method, url, data=data, headers)
predicate = lambda response: get_status_code(response) not in RETRYABLE
return _retry(func, predicate)
``